tests/unit/Metadata/MetadataSerializerTest.php

```diff
@@ -90,4 +91,12 @@ public function deserializes_serialized_data_to_same_representation(): void
 
         self::assertSame('bar', $deserializedMetadata->get('foo'));
     }
+
+    #[Test]
+    public function deserializing_throws_if_contains_invalid_data(): void
+    {
+        $this->expectExceptionObject(MetadataNotDeserializable::dueTo(new \InvalidArgumentException()));
```

Comment:
IIRC nested exception does not matter here, maybe try-catch and assert that prev is expected - if we care about such details 🙈

Reply:
I've added a test for serialization, but perhaps this simply highlights a lack of input validation as the serialization failure cannot happen with valid metadata. Let me know what you think, can improve input validation instead and remove the serialization exception entirely.

Reply:
Changed to try/catch.